doc/guides/networking/net-stack-architecture.rst

```diff
@@ -0,0 +1,137 @@
+.. _network_stack_architecture:
+
+Network Stack Architecture
+##########################
+
+High level overview of the network stack
+****************************************
+
+.. figure:: zephyr_netstack_overview.svg
+    :align: left
+    :alt: Overview of the network stack architecture
+    :figclass: align-center
+
+    Network stack overview
+
+*The network stack is layered and it consists of the following parts:*
+
+* **Network Application.** The network application uses
```

Comment:
> Now would be a good time to suggest any tweaks ...

Ok, I would suggest to change this too "The network application can either use provided application-level protocol libraries or access BSD Socket API directly."